From 82f29016519385c05ee261ec42c7d9e11044c075 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:21 -0400
Subject: [ethernet] i40e: disable local loopback on vmdq vsi

Message-id: <1409658042-22689-261-git-send-email-sassmann@redhat.com>
Patchwork-id: 90461
O-Subject: [RHEL7.1 PATCH 260/281] i40e: disable local loopback on vmdq vsi
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

The local loopback should only be enabled for VSIs that are supporting
cascaded VEBs or VEPA setups.  This is not the case here, and we need
to stop the VEB from echoing the VMDQ VSI packets back at the VSI.

Change-ID: I9dfb6ac79db24d04360d7efde62d81e20abc5090
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 738abbac9b8bf11a7cc3955b691cca8d7589127a)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 07954ec..ce30a94 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -7576,7 +7576,6 @@ static int i40e_add_vsi(struct i40e_vsi *vsi)
    * should be set to zero by default.
    */
   ctxt.info.switch_id = 0;
-  ctxt.info.switch_id |= cpu_to_le16(I40E_AQ_VSI_SW_ID_FLAG_LOCAL_LB);
   ctxt.info.switch_id |= cpu_to_le16(I40E_AQ_VSI_SW_ID_FLAG_ALLOW_LB);
 
   /* Setup the VSI tx/rx queue map for TC0 only for now */
-- 
1.7.1